### ISSUE TITLE
Panics in `logify()` with tracing enabled

### ISSUE THREAD

**Issue by hkratz** [launchdarkly/rust-eventsource-client#69]
The offending code is https://github.com/launchdarkly/rust-eventsource-client/blob/b6e61d6bed5d7733666a351ae5d5359bd879cdee/eventsource-client/src/event_parser.rs#L82-L89

This function fails in the `else` case because indexing a `&str` on a range panics if the endpoints are in the middle of multi-byte codepoints.

I replaced it locally with:
```rust
fn logify(mut bytes: &[u8]) -> &str {
    if bytes.len() > LOGIFY_MAX_CHARS {
        bytes = &bytes[..LOGIFY_MAX_CHARS];
    }
    from_utf8(bytes).unwrap_or("<bad utf8>")
}
```

**Comment by keelerm84** [comment]
Thank you for bringing this to our attention! I have put up a PR (https://github.com/launchdarkly/rust-eventsource-client/pull/70) with a suggested fix. I will let you know once we have released this.

**Comment by keelerm84** [comment]
Fixed in v0.12.1
